packages/nx/src/command-line/init/init-v2.ts

```diff
@@ -0,0 +1,171 @@
+  }
+  if (existsSync('package.json') && !options.useDotNxInstallation) {
+    const repoRoot = process.cwd();
+    const cacheableOperations: string[] = [];
+    createNxJsonFile(repoRoot, [], cacheableOperations, {});
+
+    const pmc = getPackageManagerCommand();
+
+    updateGitIgnore(repoRoot);
+
+    const plugins = await detectPlugins();
+
+    const useNxCloud =
+      options.nxCloud ??
+      (options.interactive ? await askAboutNxCloud() : false);
+
+    addDepsToPackageJson(repoRoot, plugins);
+
+    output.log({ title: '📦 Installing Nx' });
+
+    runInstall(repoRoot, pmc);
+
+    if (plugins) {
+      output.log({ title: '🔨 Configuring plugins' });
+      for (const plugin of plugins) {
+        execSync(
+          `${pmc.exec} nx g ${plugin}:init --skipPackageJson --no-interactive`,
+          {
+            stdio: [0, 1, 2],
+            cwd: repoRoot,
+          }
+        );
+      }
+    }
+
+    if (useNxCloud) {
+      output.log({ title: '🛠️ Setting up Nx Cloud' });
+      execSync(
+        `${pmc.exec} nx g nx:connect-to-nx-cloud --installationSource=nx-init-pcv3 --quiet --no-interactive`,
+        {
+          stdio: [0, 1, 2],
+          cwd: repoRoot,
+        }
+      );
+    }
+  } else {
+    if (process.platform !== 'win32') {
+      console.log(
+        'Setting Nx up installation in `.nx`. You can run nx commands like: `./nx --help`'
+      );
+    } else {
+      console.log(
+        'Setting Nx up installation in `.nx`. You can run nx commands like: `./nx.bat --help`'
+      );
+    }
+    generateDotNxSetup(version);
+    // invokes the wrapper, thus invoking the initial installation process
+    runNxSync('');
+  }
+}
+
+async function detectPlugins(): Promise<undefined | string[]> {
+  const files = ['package.json'].concat(
+    sync(`**/*/package.json`, {
+      ignore: ['**/node_modules/**', '**/build/**', '**/dist/**', '**/tmp/**'],
+    })
+  );
+  const detectedPlugins = new Set<string>();
+  for (const file of files) {
+    if (!existsSync(file)) continue;
+
+    let packageJson: PackageJson;
+    try {
+      packageJson = readJsonFile(file);
+    } catch {
+      // Could have malformed JSON for unit tests, etc.
+      continue;
+    }
+
+    const deps = {
+      ...packageJson.dependencies,
+      ...packageJson.devDependencies,
+    };
+
+    // Generic JS tools
+    if (deps['eslint']) {
+      detectedPlugins.add('@nx/eslint');
+    }
+    // Bundlers
+    if (deps['vite'] || deps['vitest']) {
+      detectedPlugins.add('@nx/vite');
+    }
+    if (deps['webpack']) {
+      detectedPlugins.add('@nx/webpack');
+    }
+    // Testing tools
+    if (deps['jest']) {
+      detectedPlugins.add('@nx/jest');
+    }
+    if (deps['cypress']) {
+      detectedPlugins.add('@nx/cypress');
+    }
+    if (deps['playwright']) {
+      detectedPlugins.add('@nx/playwright');
+    }
+    // Frameworks
+    if (deps['@nestjs/core'] && deps['@nestjs/cli']) {
```

Comment:
Do we have a PCV3 `createNodes` plugin for Nest?

Reply:
Won't be ready, I'll remove for now.